pylint/test/unittest_lint.py

```diff
@@ -232,6 +232,20 @@ def init_linter(linter):
     return linter
 
 
+def test_register_checker_type_once():
+    class CustomChecker(checkers.BaseChecker):
+        name = 'custom'
+
+    linter = PyLinter()
+    plugin_registry = PluginRegistry(linter)
+    checker1 = CustomChecker()
+    checker2 = CustomChecker()
+    plugin_registry.register_checker(checker1)
+    plugin_registry.register_checker(checker2)
+
+    assert list(plugin_registry.for_all_checkers()) == [linter, checker1]
```

Comment:
linter is added to checkers in init. If that changes, this test will need updating. Using public method for_all_checkers instead of private _checkers for test.

Reply:
So this did need changing after the recent changes, good guess!